src/renderer.cpp

```diff
@@ -1357,6 +1750,248 @@ void Renderer::setD3DProjection(struct matrix* matrix)
     if (uniform_log) printMatrix(__func__, internalState.d3dProjectionMatrix);
 };
 
+void Renderer::setWorldLightDir(float dirX, float dirY, float dirZ)
+{
+    worldLightRot.x = dirX;
+    worldLightRot.y = dirY;
+    worldLightRot.z = dirZ;
+}
+
+struct point3d Renderer::getWorldLightDir()
+{
+    return worldLightRot;
+}
+
+void Renderer::setLightIntensity(float intensity)
+{
+    internalState.lightData[3] = intensity;
+}
+
+float Renderer::getLightIntensity()
+{
+    return internalState.lightData[3];
+}
+
+void Renderer::setLightColor(float r, float g, float b)
+{
+    internalState.lightData[0] = r;
+    internalState.lightData[1] = g;
+    internalState.lightData[2] = b;
+}
+
+point3d Renderer::getLightColor()
+{
+    struct point3d color = { internalState.lightData[0],
+                             internalState.lightData[1],
+                             internalState.lightData[2] };
+    return color;
+}
+
+void Renderer::setAmbientIntensity(float intensity)
+{
+    internalState.ambientLightData[3] = intensity;
+}
+
+float Renderer::getAmbientIntensity()
+{
+    return internalState.ambientLightData[3];
+}
+
+void Renderer::setAmbientLightColor(float r, float g, float b)
+{
+    internalState.ambientLightData[0] = r;
+    internalState.ambientLightData[1] = g;
+    internalState.ambientLightData[2] = b;
+}
+
+point3d Renderer::getAmbientLightColor()
+{
+    struct point3d color = { internalState.ambientLightData[0],
+                             internalState.ambientLightData[1],
+                             internalState.ambientLightData[2] };
+    return color;
+}
+
+void Renderer::setRoughness(float roughness)
+{
+    internalState.materialData[0] = roughness;
+}
+
+float Renderer::getRoughness()
+{
+    return internalState.materialData[0];
+}
+
+void Renderer::setMetalness(float metalness)
+{
+    internalState.materialData[1] = metalness;
+}
+
+float Renderer::getMetalness()
+{
+    return internalState.materialData[1];
+}
+
+void Renderer::setShadowFaceCullingEnabled(bool isEnabled)
+{
+    isShadowMapFaceCullingEnabled = isEnabled;

```

Comment:
Shouldn't this be part of the `internalState`?

Reply:
I moved this variable to the `internalState`.